src/commands/crashes/upload-mappings.ts

```diff
@@ -0,0 +1,88 @@
+import { AppCenterClient, models } from "../../util/apis";
+import { AppCommand, CommandResult } from "../../util/commandline";
+import { ErrorCodes, failure, success } from "../../util/commandline";
+import { hasArg, help, longName, shortName } from "../../util/commandline";
+import { out } from "../../util/interaction";
+import { inspect } from "util";
+import { DefaultApp } from "../../util/profile";
+import UploadSymbolsHelper from "./lib/symbols-uploading-helper";
+import { SymbolType } from "./lib/symbols-uploading-helper";
+
+import * as Fs from "fs";
+import * as Path from "path";
+
+import * as _ from "lodash";
+
+const debug = require("debug")("appcenter-cli:commands:apps:crashes:upload-mappings");
+
+@help("Upload the Android mappings for the application")
+export default class UploadMappings extends AppCommand {
+  @help("Path to an Android mapping.txt file.")
+  @shortName("m")
+  @longName("mapping")
+  @hasArg
+  public mappingPath: string;
+
+  @help("The version name to associate with the mappings.")
+  @shortName("n")
+  @longName("version-name")
+  @hasArg
+  public versionName: string;
+
+  @help("The version code to associate with the mappings.")
+  @shortName("c")
+  @longName("version-code")
+  @hasArg
+  public versionCode: string;
+
+  public async run(client: AppCenterClient): Promise<CommandResult> {
+    const app: DefaultApp = this.app;
+
+    this.validateParameters();
+
+    this.getStatsForFsPath(this.mappingPath);
+    if (this.getLowerCasedFileExtension(this.mappingPath) !== ".txt") {
```

Comment:
I think checking for the entire file name might be a bit restrictive, especially since the CLI is also executed on a developer machine, not only in the concept of one particular build which creates one mapping.txt file - so the user might have assembled a different structure, i.e. a folder where they keep multiple mapping.txt files for different versions like "mapping-1.0-21.txt" etc.